test/eventHandlers/invocationRequest.test.ts

```diff
@@ -438,8 +438,8 @@ describe('invocationRequest', () => {
         sendInvokeMessage([httpInputData]);
         await stream.assertCalledWith(
             Msg.receivedInvocLog(),
-            Msg.invocResponse([getHttpResponse()]),
-            Msg.asyncAndDoneLog
+            Msg.asyncAndDoneLog,

```

Comment:
> What are the the logs changes? Only category change?

The only other change has to do with ordering/timing of the logs (that's why these lines are switched in the test). **tl;dr** if the user is using context.done and executing synchronous code after they call context.done the logs will be in a different order. Didn't seem like a big deal because the same exact logs are still displayed and to get into this situation the user is doing a bunch of stuff we explicitly tell them not to do.

Here's a more in-depth analysis if you're curious:

If the user has _synchronous blocking_ code after they call `context.done`, the invocation response will now come after that code finishes. Here's an example:

```typescript
module.exports = function (context: Context, _req: HttpRequest): void {
    context.log('1');
    delaySync(500);
    context.log('2');
    context.done();
    delaySync(500);
    context.log('3');
};

function delaySync(ms: number) {
    const end = Date.now() + ms;
    while (Date.now() < end) {
        // wait
    }
}
```

### Previous Behavior

```
[2022-03-08T01:00:14.830Z] Executing 'Functions.delay' (Reason='This function was programmatically called via the host APIs.', Id=cb92b92f-aa3a-4154-bb0a-66f734a00ca1)
[2022-03-08T01:00:15.916Z] 1
[2022-03-08T01:00:15.916Z] 2
[2022-03-08T01:00:15.929Z] InvocationResponse received for invocation id: 'cb92b92f-aa3a-4154-bb0a-66f734a00ca1'
[2022-03-08T01:00:16.004Z] Executed 'Functions.delay' (Succeeded, Id=cb92b92f-aa3a-4154-bb0a-66f734a00ca1, Duration=1212ms)
[2022-03-08T01:00:16.031Z] Warning: Unexpected call to 'log' on the context object after function execution has completed. Please check for asynchronous calls that are not awaited or calls to 'done' made before function execution completes. Function name: delay. Invocation Id: cb92b92f-aa3a-4154-bb0a-66f734a00ca1. Learn more: https://go.microsoft.com/fwlink/?linkid=2097909 
```

### New Behavior

```
[2022-03-08T01:09:20.582Z] Executing 'Functions.delay' (Reason='This function was programmatically called via the host APIs.', Id=1237a61d-c126-4d0a-8c93-5821409fae84)
[2022-03-08T01:09:21.587Z] 1
[2022-03-08T01:09:21.587Z] 2
[2022-03-08T01:09:21.587Z] Warning: Unexpected call to 'log' on the context object after function execution has completed. Please check for asynchronous calls that are not awaited or calls to 'done' made before function execution completes. Function name: delay. Invocation Id: 1237a61d-c126-4d0a-8c93-5821409fae84. Learn more: https://go.microsoft.com/fwlink/?linkid=2097909 
[2022-03-08T01:09:21.587Z] 3
[2022-03-08T01:09:21.588Z] InvocationResponse received for invocation id: '1237a61d-c126-4d0a-8c93-5821409fae84'
[2022-03-08T01:09:21.588Z] Executed 'Functions.delay' (Succeeded, Id=1237a61d-c126-4d0a-8c93-5821409fae84, Duration=1006ms)
```

I think people doing that kind of sync blocking code after context.done are very rare. Any _non-blocking_ code (like a callback or async method) will have the exact same behavior as before, even if I use 1ms as the timeout like this:
```typescript
module.exports = function (context: Context, _req: HttpRequest): void {
    context.log('1');
    setTimeout(() => {
        context.log('2');
        context.done();
        setTimeout(() => {
            context.log('3');
        }, 1);
    }, 1);
};
```

Reply:
it looks like a breaking change: `context.log('3');` is not executed in previous behavior, we exit from function code immediately. With new behavior we continue executing even after context.done(). Imagine if the customer has the code after `context.done`. It can break the logic.

```
if ()
{
  context.done();
}
// other code 
```

Reply:
Thanks for the clarification. Also we emit log in new behavior after `context.done` but I think it's not a very big deal.